# Sustainability challenge

---

## The brief

> **URGENT: Sustainability team request**
>
> The sustainability team is presenting to leadership Monday morning. They need a quick data snapshot of our product assortment's sustainability profile before EOD.
>
> Use the Systembolaget data to answer their questions below.
>
> **Present your findings clearly** - they're not technical people. Use markdown cells to write your conclusions.
>
> **Deadline: EOD**

---

## Getting started

Run this cell to load the data:

In [76]:
import requests
import pandas as pd

response = requests.get("https://susbolaget.emrik.org/v1/products")
data = response.json()
df = pd.DataFrame(data)

print(f"Loaded {len(df)} products")

Loaded 27364 products


### Useful columns for this assignment:

| Column | Description |
|--------|-------------|
| `isOrganic` | True/False - is the product organic? |
| `isSustainableChoice` | True/False - Systembolaget's sustainable choice label |
| `isClimateSmartPackaging` | True/False - climate-smart packaging |
| `isEthical` | True/False - ethical certification |
| `categoryLevel1` | Main category (Vin, Öl, Sprit, etc.) |
| `country` | Country of origin |
| `price` | Price in SEK |


<details>
<summary><strong>Click here for full column reference</strong></summary>

### Product identification
| Column | Type | Description |
|--------|------|-------------|
| `productId` | string | Unique product identifier |
| `productNumber` | string | Product number (full) |
| `productNumberShort` | string | Short product number |
| `productNameBold` | string | Product name (brand/main name) |
| `productNameThin` | string | Product name (variant/description) |

### Producer & Supplier
| Column | Type | Description |
|--------|------|-------------|
| `producerName` | string | Name of the producer |
| `supplierName` | string | Name of the supplier |

### Categories
| Column | Type | Description |
|--------|------|-------------|
| `category` | string | General category |
| `categoryLevel1` | string | Main category (Vin, Öl, Sprit, Cider & blanddrycker, Alkoholfritt) |
| `categoryLevel2` | string | Sub-category (e.g., Rött vin, Vitt vin, Lager) |
| `categoryLevel3` | string | More specific category |
| `categoryLevel4` | string | Most specific category |
| `customCategoryTitle` | string | Custom display category |

### Origin
| Column | Type | Description |
|--------|------|-------------|
| `country` | string | Country of origin |
| `originLevel1` | string | Region level 1 |
| `originLevel2` | string | Region level 2 (more specific) |
| `isManufacturingCountry` | bool | Is it manufactured in the listed country? |

### Price & Volume
| Column | Type | Description |
|--------|------|-------------|
| `price` | float | Price in SEK |
| `volume` | int | Volume in ml |
| `volumeText` | string | Volume as text (e.g., "750 ml") |
| `alcoholPercentage` | float | Alcohol percentage |
| `recycleFee` | float | Recycling fee (pant) |

### Sustainability flags
| Column | Type | Description |
|--------|------|-------------|
| `isOrganic` | bool | Organic certification |
| `isSustainableChoice` | bool | Systembolaget's sustainable choice |
| `isClimateSmartPackaging` | bool | Climate-smart packaging |
| `isEthical` | bool | Ethical certification |
| `ethicalLabel` | string | Name of ethical certification |

### Packaging
| Column | Type | Description |
|--------|------|-------------|
| `bottleText` | string | Bottle/container type (Flaska, Burk, etc.) |
| `packagingLevel1` | string | Packaging category |
| `packagingCO2ImpactLevel` | string | CO2 impact level (Lägre, Medel, Högre) |

### Taste characteristics
| Column | Type | Description |
|--------|------|-------------|
| `taste` | string | Taste description (free text) |
| `tasteSymbols` | list | Taste symbols/pairings |
| `tasteClockSweetness` | int | Sweetness (0-12 scale) |
| `tasteClockBody` | int | Body/fullness (0-12 scale) |
| `tasteClockFruitacid` | int | Fruit acidity (0-12 scale) |
| `tasteClockBitter` | int | Bitterness (0-12 scale) |
| `tasteClockRoughness` | int | Roughness/tannins (0-12 scale) |
| `tasteClockSmokiness` | int | Smokiness (0-12 scale) |
| `tasteClockCasque` | int | Oak/barrel character (0-12 scale) |
| `color` | string | Color description |
| `usage` | string | Serving suggestion |

### Wine-specific
| Column | Type | Description |
|--------|------|-------------|
| `vintage` | string | Vintage year |
| `grapes` | list | Grape varieties |
| `seal` | string | Closure type (cork, screw cap, etc.) |

### Stock & Availability
| Column | Type | Description |
|--------|------|-------------|
| `isCompletelyOutOfStock` | bool | Completely out of stock |
| `isTemporaryOutOfStock` | bool | Temporarily out of stock |
| `isDiscontinued` | bool | Product discontinued |
| `assortmentText` | string | Assortment type (Fast, Tillfälligt, etc.) |
| `assortment` | string | Assortment code |

### Launch & News
| Column | Type | Description |
|--------|------|-------------|
| `isNews` | bool | Is it a new product? |
| `isWebLaunch` | bool | Web launch product |
| `productLaunchDate` | string | Launch date (ISO format) |

### Historical data (nested)
| Column | Type | Description |
|--------|------|-------------|
| `priceHistory` | list | List of {x: timestamp, y: price} |
| `alcoholHistory` | list | List of {x: timestamp, y: alcohol%} |
| `soldVolume` | list | List of {x: timestamp, y: volume sold} |

### Other
| Column | Type | Description |
|--------|------|-------------|
| `isKosher` | bool | Kosher certification |
| `sugarContent` | float | Sugar content (g/l) |
| `sugarContentGramPer100ml` | float | Sugar per 100ml |
| `images` | list | Product images |

</details>

---

## 1. The big picture

*What percentage of our assortment is organic? What about sustainable choice and climate-smart packaging? Etc.*

In [80]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27364 entries, 0 to 27363
Data columns (total 83 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   productId                        27364 non-null  object 
 1   productNumber                    27364 non-null  object 
 2   productNameBold                  27364 non-null  object 
 3   productNameThin                  24506 non-null  object 
 4   category                         0 non-null      object 
 5   productNumberShort               27364 non-null  object 
 6   producerName                     27291 non-null  object 
 7   supplierName                     27356 non-null  object 
 8   isKosher                         27364 non-null  bool   
 9   bottleText                       27364 non-null  object 
 10  restrictedParcelQuantity         27364 non-null  int64  
 11  isOrganic                        27364 non-null  bool   
 12  isSustainableChoic

In [7]:
type(df)

pandas.core.frame.DataFrame

In [108]:
all_org_counts = df["isOrganic"].value_counts()
org_count = all_org_counts.get(True, 0)
total_org_count = all_org_counts.sum()
organic_percentage = (org_count / total_org_count) * 100 

print(f"The percentage of organic products is: {organic_percentage: .2f}%")

The percentage of organic products is:  12.34%


In [109]:
all_eth_counts = df["isEthical"].value_counts()
eth_count = all_eth_counts.get(True, 0)
total_eth_count = all_eth_counts.sum()
ethical_percentage = (eth_count / total_eth_count) * 100 

print(f"The percentage of ethical products is : {ethical_percentage: .2f}%")

The percentage of ethical products is :  0.35%


In [123]:
all_sust_counts = df["isSustainableChoice"].value_counts()
sust_count = all_sust_counts.get(True, 0)
total_sust_count = all_sust_counts.sum()
sust_percentage = (sust_count / total_sust_count) * 100

print(f"The percentage of sustainable products is : {sust_percentage: .2f}%")

The percentage of sustainable products is :  1.66%


In [128]:
impact_percentages = df["packagingCO2ImpactLevel"].value_counts(normalize=True).round(2) * 100
for category, percentage in impact_percentages.items():
    print(f"The packaging CO2 impact level '{category}' is: {percentage:.2f}%")

The packaging CO2 impact level 'Högre' is: 75.00%
The packaging CO2 impact level 'Lägre' is: 13.00%
The packaging CO2 impact level 'Medel' is: 12.00%


Percentages: 
- Organic products: 12.34%
- Ethical products: 0.35%
- Sustainable products: 1.66%

The percentages are quite low for a company that wants to focus on sustainaibility, especially the one for "ethical products".
At the same time we have a 75% on the "Högre" category of the CO2 impact level, a number way too high. 

---

## 2. Category breakdown

*Which product categories have the highest percentage of organic products? Which have the lowest? Other intereting findings?*

In [176]:
product_category = df.groupby("categoryLevel1")["isOrganic"].mean().sort_values(ascending=False)

print(product_category)
print(f"The product category with the highest percentage of organic product is {product_category.index[0]}")
print(f"The product category with the lowest percentage of organic product is {product_category.index[-1]}")

categoryLevel1
Vin                     0.175781
Alkoholfritt            0.167488
Cider & blanddrycker    0.151452
Sprit                   0.048832
Öl                      0.032385
Name: isOrganic, dtype: float64
The product category with the highest percentage of organic product is Vin
The product category with the lowest percentage of organic product is Öl


**Your findings:**

*Write your conclusions here...*

---

## 3. The price question

*Leadership will ask: "Do organic products cost more?"*

*Compare the average price of organic vs non-organic products. What's the difference? Has it changed over time?*

In [193]:
avg_price = df.groupby('isOrganic')['price'].mean()
non_organic = avg_price.loc[False]
organic = avg_price.loc[True]
difference = non_organic - organic 


print(f"Non-Organic Average Price: {non_organic:.2f} SEK")
print(f"Organic Average Price: {organic:.2f} SEK")
print(f"Difference: {difference:.2f} SEK")

Non-Organic Average Price: 657.68 SEK
Organic Average Price: 378.79 SEK
Difference: 278.90 SEK


In [215]:
row = df.iloc[0]
price_history = row["priceHistory"]

def first_value(price_history):
    return price_history[0]["y"]

avg_first_price = df.groupby('isOrganic')['firstPrice'].mean()
non_org_first_value = avg_first_price.loc[False]
org_first_value = avg_first_price.loc[True]
first_value_diff = non_org_first_value - org_first_value

print(f"Original Non-Organic Average Price: {non_org_first_value:.2f} SEK")
print(f"Originla Organic Average Price: {org_first_value:.2f} SEK")
print(f"Original difference: {first_value_diff:.2f} SEK")

Original Non-Organic Average Price: 632.98 SEK
Originla Organic Average Price: 375.03 SEK
Original difference: 257.95 SEK


In [223]:
print(non_organic - non_org_first_value) 
print(organic - org_first_value)

24.70791961977818
3.760568383658949


Averege price: 
- Non-Organic: 657.68 SEK
- Organic: 378.79 SEK
- Difference: 278.90

Original average price: 
- Non-Organic: 632.98
- Organic: 375.03
- Difference: 257.95

The prices of non-organic products increases, in time, more than the prices of organic products. 
A possible reason could be that spirits has in general higher prices and this category tend to have a very low percentage of organic products. 

---

## 4. Origin analysis

*Which countries produce the most organic products in our assortment? What origins are popular? Does it differ for categories? And is it different from non-organic products?*

In [233]:
df[df["isOrganic"] == True]["country"].value_counts().head(5)

country
Frankrike    898
Italien      781
Spanien      482
Sverige      435
Österrike    192
Name: count, dtype: int64

In [234]:
df[df["isOrganic"] == True]["originLevel1"].value_counts().head(5)

originLevel1
Toscana       202
Bourgogne     158
Rhonedalen    135
Katalonien    119
Piemonte      115
Name: count, dtype: int64

In [235]:
df[df["isOrganic"] == True]["originLevel2"].value_counts().head(5)

originLevel2
Chianti Classico    61
Penedès             59
Côte de Nuits       50
Côtes du Rhône      43
Touraine            42
Name: count, dtype: int64

In [236]:
df[df["isOrganic"] == False]["country"].value_counts().head(5)

country
Sverige           5207
Frankrike         4357
Italien           3669
Storbritannien    1699
Spanien           1521
Name: count, dtype: int64

In [237]:
df[df["isOrganic"] == False]["originLevel1"].value_counts().head(5)

originLevel1
Champagne               1214
Skottland               1203
Piemonte                1151
Västra Götalands län     947
Bourgogne                873
Name: count, dtype: int64

In [239]:
df[df["isOrganic"] == False]["originLevel2"].value_counts().head(5)

originLevel2
Highlands         530
Göteborgs stad    393
Barolo            385
Coastal Region    332
Côte de Beaune    226
Name: count, dtype: int64

Top 5 countries that produce organic products:
- France        898
- Italy         781
- Spain         482
- Sweden        435 
- Austria       192

4 out of 5 of those countries are also in the list of top 5 producers of non-organic products, that makes them the leaders in the general market.

Top 5 regions: 
- Toscana       202
- Bourgogne     158
- Rhonedalen    135
- Katalonien    119
- Piemonte      115


Non-organic product are still the most produced, the difference in numbers is pretty significant. 
That means that the request in the market of non-organic products is still really high.

## 5. Overlap analysis

*How many products are both organic AND sustainable choice? How many are organic OR sustainable choice? How does the overlaps look like for different categories?*

In [248]:
org_and_sust = df[(df["isOrganic"] == True) & (df["isSustainableChoice"] == True)]

print(f"{len(org_and_sust)} is the number of products that are both organic and sustainable")

260 is the number of products that are both organic and sustainable


In [249]:
org_or_sust = df[(df["isOrganic"] == True) | (df["isSustainableChoice"] == True)]

print(f"{len(org_or_sust)} is the number of products that are organic or sustainable")

3573 is the number of products that are organic or sustainable


---

## 5. Bonus challenge

**Sustainability score:** Create a new column that counts how many sustainability flags are True for each product (organic, sustainable choice, climate-smart packaging, ethical). Which products score highest?

---

## Summary for leadership

*Use this cell to write a 3-5 bullet point executive summary of your findings:*

- 
- 
- 